extensions/wikia/TemplateClassification/services/UserTemplateClassificationService.class.php

```diff
@@ -43,4 +50,23 @@ public function getType( $wikiId, $pageId ) {
 
 		return $templateType;
 	}
+
+	/**
+	 * Verify template type before user classification
+	 *
+	 * @param int $wikiId
+	 * @param int $pageId
+	 * @param string $templateType
+	 * @param string $origin
+	 * @throws BadRequestApiException
+	 */
+	public function classifyTemplate( $wikiId, $pageId, $templateType, $origin ) {
+		if ( !in_array( $templateType, self::$templateTypes ) ) {
+			throw new ApiException( self::CLASSIFY_TEMPLATE_EXCEPTION_MESSAGE, self::CLASSIFY_TEMPLATE_EXCEPTION_CODE );
+		}
+
+		parent::classifyTemplate( $wikiId, $pageId, $templateType, self::USER_PROVIDER, $origin );
+
+		wfRunHooks( 'UserTemplateClassification::TemplateClassified', [ $wikiId, $pageId, $templateType ] );
```

Comment:
From performance perspective we can have a gain if we create it once before hook as each call of Title::newFromID generates DB query
https://github.com/Wikia/app/blob/CE-3110/includes/Title.php#L226-226
plus there's small possibility of having all those hooks ever removed...